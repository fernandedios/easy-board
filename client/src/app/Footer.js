import React from 'react';

const Footer = () => {
    return (
        <footer className="footer">
            <div className="container">
                <div className="content has-text-centered">
                    <p>&copy; Easy Board. Open Source under MIT License.</p>
                </div>
            </div>
        </footer>
    );
};

export default Footer;
