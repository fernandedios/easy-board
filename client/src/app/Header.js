import React, { Component } from 'react';

export class Header extends Component {
    constructor(props) {
        super(props);

        this.toggleMenu = this.toggleMenu.bind(this);
    }

    toggleMenu() {
        const navBurger = document.querySelector('.navbar-burger');
        const navMenu = document.querySelector('#navbarMain');
        navBurger.classList.toggle('is-active');
        navMenu.classList.toggle('is-active');
    }

    render() {
        return (
            <nav className="navbar is-dark">
                <div className="navbar-brand">
                    <a className="navbar-item" href="/">
                        <img src="/img/logo.png" alt="Easy Board Collaboration Tool" height="28" /> Easy Board
                    </a>
                    <div onClick={this.toggleMenu} className="navbar-burger burger" data-target="navbarMain">
                        <span></span>
                        <span></span>
                        <span></span>
                    </div>
                </div>

                <div id="navbarMain" className="navbar-menu">
                    <div className="navbar-end">
                        <a className="navbar-item" href="https://bulma.io/">Home</a>
                        <a className="navbar-item" href="https://bulma.io/">Home</a>
                        <a className="navbar-item" href="https://bulma.io/">Home</a>
                        <a className="navbar-item" href="https://bulma.io/">Home</a>
                    </div>
                </div>
            </nav>
        );
    }
}

export default Header;
