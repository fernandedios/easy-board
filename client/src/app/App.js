import '../index.css';
import React, { Component } from 'react';

import Header from './Header';
import Footer from './Footer';
import Login from '../login/Login';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <div className="main-content">
          <Login />
        </div>
        <Footer />
      </div>
    );
  }
}

export default App;
