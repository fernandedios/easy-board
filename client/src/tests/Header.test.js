import React from 'react';
import { shallow, mount } from 'enzyme';
import ReactTestRenderer from 'react-test-renderer';

import { Header } from '../app/Header';

describe('<Header />', () => {
    it('should be defined', () => {
        expect(Header).toBeDefined();
    });
    
    it('should have correct classNames', () => {
        const output = shallow(<Header />);
        expect(output.find('nav').hasClass('navbar')).toEqual(true);
    });

    it('should render menu', () => {
        const output = mount(<Header />);
        expect(output.find('.navbar-end').children()).toHaveLength(4);
    });

    it('should render correctly', () => {
        const output = ReactTestRenderer.create(<Header />).toJSON();
        expect(output).toMatchSnapshot();
    });
});
