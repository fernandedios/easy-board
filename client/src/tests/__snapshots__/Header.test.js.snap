// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<nav
  className="navbar is-dark"
>
  <div
    className="navbar-brand"
  >
    <a
      className="navbar-item"
      href="/"
    >
      <img
        alt="Easy Board Collaboration Tool"
        height="28"
        src="/img/logo.png"
      />
       Easy Board
    </a>
    <div
      className="navbar-burger burger"
      data-target="navbarMain"
      onClick={[Function]}
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="navbar-menu"
    id="navbarMain"
  >
    <div
      className="navbar-end"
    >
      <a
        className="navbar-item"
        href="https://bulma.io/"
      >
        Home
      </a>
      <a
        className="navbar-item"
        href="https://bulma.io/"
      >
        Home
      </a>
      <a
        className="navbar-item"
        href="https://bulma.io/"
      >
        Home
      </a>
      <a
        className="navbar-item"
        href="https://bulma.io/"
      >
        Home
      </a>
    </div>
  </div>
</nav>
`;
